streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -668,8 +669,10 @@ public synchronized void onChange(final Thread thread,
                     if (newState == GlobalStreamThread.State.RUNNING) {
                         maybeSetRunning();
                     } else if (newState == GlobalStreamThread.State.DEAD) {
-                        log.error("Global thread has died. The streams application or client will now close to ERROR.");
-                        closeToError();
+                        if (state != State.PENDING_SHUTDOWN) {
```

Comment:
New solution: if the instance is not in `PENDING_SHUTDOW` we know that an error happened and we log an ERROR and `closeToError`.

On a clean close, the instance with go to `PENDING_SHUTDOWN` first, before shutting down the global thread, and we won't log an error and just proceed.

Reply:
Thanks @mjsax , I think this works too.